pkg/action/failback.go

```diff
@@ -0,0 +1,216 @@
+			log.ActionLog(action).Infof(msg)
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusScheduled),
+				msg)
+			ac.updateAction(action)
+			return
+		}
+
+		if !utils.IsSubList(namespaces, action.Spec.ActionParameter.FailbackParameter.Namespaces) {
+			msg := fmt.Sprintf("Namespaces provided for failback is not a subset of namespaces from migrationschedule %s", migrationSchedule.Name)
+			log.ActionLog(action).Infof(msg)
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusScheduled),
+				msg)
+			ac.updateAction(action)
+			return
+		}
+
+	}
+
+	// Return if any migration is ongoing
+	for _, policyType := range storkv1.GetValidSchedulePolicyTypes() {
+		policyMigration, present := migrationSchedule.Status.Items[policyType]
+		if present {
+			for _, migration := range policyMigration {
+				if !isMigrationComplete(migration.Status) {
+					msg := fmt.Sprintf("Waiting for the completion of migration %s", migration.Name)
+					log.ActionLog(action).Infof(msg)
+					ac.recorder.Event(action,
+						v1.EventTypeWarning,
+						string(storkv1.ActionStatusScheduled),
+						msg)
+					ac.updateAction(action)
+					return
+				}
+			}
+		}
+	}
+
+	// check if it has any latest successful migration
+	schedulePolicyType, latestMigration := getLatestMigrationPolicyAndStatus(*migrationSchedule)
+	if latestMigration.Status == storkv1.MigrationStatusFailed {
+		msg := fmt.Sprintf("The latest migration %s is in %s state, hence aborting failback", latestMigration.Name, storkv1.MigrationStatusFailed)
+		log.ActionLog(action).Errorf(msg)
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusFailed),
+			msg)
+		ac.updateAction(action)
+		return
+	}
+
+	if latestMigration.Status == storkv1.MigrationStatusSuccessful ||
+		latestMigration.Status == storkv1.MigrationStatusPartialSuccess {
+		// Check if the latest migration has run with in the policy time
+		alreadyTriggered, err := schedule.AlreadyTriggered(
+			migrationSchedule.Spec.SchedulePolicyName,
+			migrationSchedule.Namespace,
+			schedulePolicyType,
+			latestMigration.FinishTimestamp,
+		)
+		if err != nil {
+			msg := fmt.Sprintf("Error while checking if the latest migration %s has been successful: %v", latestMigration.Name, err)
+			log.ActionLog(action).Errorf(msg)
+			action.Status.Status = storkv1.ActionStatusFailed
+			action.Status.Reason = msg
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusFailed),
+				msg)
+			ac.updateAction(action)
+			return
+		}
+		if alreadyTriggered {
+			log.ActionLog(action).Infof("The latest migration %s is in threshold range", latestMigration.Name)
+			// Suspend migration schedule
+			log.ActionLog(action).Infof("Suspending the migration schedule %s", migrationSchedule.Name)
```

Comment:
nit: i don't think this particular log is needed since we have a similar one in the below line#135

Reply:
Done